library/java/src/io/envoyproxy/envoymobile/engine/EnvoyConfiguration.java

```diff
@@ -1,11 +1,46 @@
 package io.envoyproxy.envoymobile.engine;
 
-public interface EnvoyConfiguration {
+public class EnvoyConfiguration {
+
+  public final Integer connectTimeoutSeconds;
+  public final Integer dnsRefreshSeconds;
+  public final Integer statsFlushSeconds;
 
   /**
-   * Provides a default configuration template that may be used for starting Envoy.
+   * Create an EnvoyConfiguration with a user provided configuration values.
    *
-   * @return A template that may be used as a starting point for constructing configurations.
+   * @param connectTimeoutSeconds timeout for new network connections to hosts in the cluster.
+   * @param dnsRefreshSeconds rate in seconds to refresh DNS.
+   * @param statsFlushSeconds interval at which to flush Envoy stats.
    */
-  String templateString();
+  public EnvoyConfiguration(int connectTimeoutSeconds, int dnsRefreshSeconds,
+                            int statsFlushSeconds) {
+    this.connectTimeoutSeconds = connectTimeoutSeconds;
+    this.dnsRefreshSeconds = dnsRefreshSeconds;
+    this.statsFlushSeconds = statsFlushSeconds;
+  }
+
+  /**
+   * Resolves the provided configuration template using properties on this configuration.
+   * This default configuration is provided by the native layer.
+   *
+   * @param templateYAML the default template configuration.
+   * @return String, the resolved template.
+   * @throws ConfigurationException, when the template provided is not fully resolved.
+   */
+  String resolveTemplate(String templateYAML) {
```

Comment:
Thoughts on moving the logic in this function into `EnvoyConfigurationImpl` (could be a helper method called out of `runWithConfig`)? That way the configuration really doesn't need to know or care about any implementation details of how its attributes are used.

Reply:
I can create another file which holds this method and just port the unit tests there -- it'll make this class a pure data class then. Wdyt?

cc: @rebello95

Reply:
Sure, I'm good with that

Reply:
Actually, let's revisit at a later date on this 😬